test/gui/shared/scripts/pageObjects/AccountConnectionWizard.py

```diff
@@ -121,33 +139,60 @@ def addServer(self, context):
         )
         squish.clickButton(squish.waitForObject(self.NEXT_BUTTON))
 
-        try:
-            squish.clickButton(
-                squish.waitForObject(self.CONFIRM_INSECURE_CONNECTION_BUTTON)
-            )
-        except:
-            test.log(
-                "No insecure connection warning for server " + clientDetails['server']
-            )
-            pass
+        if not context.userData['ocis']:
+            try:
+                squish.clickButton(
+                    squish.waitForObject(self.CONFIRM_INSECURE_CONNECTION_BUTTON, 1000)
+                )
+            except:
+                test.log(
+                    "No insecure connection warning for server "
+                    + clientDetails['server']
+                )
+                pass
+
+    def acceptCertificate(self):
+        squish.clickButton(squish.waitForObject(self.ACCEPT_CERTIFICATE_YES))
 
     def addUserCreds(self, context):
         clientDetails = getClientDetails(context)
 
+        if context.userData['ocis']:
+            self.oidcLogin(clientDetails['user'], clientDetails['password'])
+        else:
+            self.basicLogin(clientDetails['user'], clientDetails['password'])
+
+    def basicLogin(self, username, password):
         squish.type(
             squish.waitForObject(self.USERNAME_BOX),
-            clientDetails['user'],
+            username,
         )
         squish.type(
             squish.waitForObject(self.USERNAME_BOX),
             "<Tab>",
         )
         squish.type(
             squish.waitForObject(self.PASSWORD_BOX),
-            clientDetails['password'],
+            password,
         )
         squish.clickButton(squish.waitForObject(self.NEXT_BUTTON))
 
+    def oidcLogin(self, username, password, relogin=False):
+        # wait 500ms for copy button to fully load
+        squish.snooze(1 / 2)
```

Comment:
why not using `waitForObject`?

Reply:
it is used below but the `squish.getClipboardText()` would crash so extra wait was required. But I will recheck it

Reply:
Couldn't find favorable solution for this one other than explicit snooze